File: csharp/Source/Estruturas/Repetidas/DentroFora.cs
namespace Source.Estruturas.Repetidas;

/**
 * Problema "dentro_fora"
 * Leia um valor inteiro N.
 * Este valor será a quantidade de valores inteiros X que serão lidos em seguida.
 * Mostre quantos destes valores X estão dentro do intervalo [10,20]
 * e quantos estão fora do intervalo.
 */
internal static class DentroFora
{
    private static readonly short LOWER_BOUND = 10;
    private static readonly short UPPER_BOUND = 20;
    private static readonly short MAX_NUMBERS = 10;

    public static void Execute()
    {
        Console.Write($"Quantos números você vai digitar (máximo {MAX_NUMBERS})? ");
        int n = int.Parse(Console.ReadLine()!);

        n = n < 1 || n > MAX_NUMBERS ? 1 : n;

        int x;
        int insideCount = 0;
        int outsideCount = 0;

        for (int i = 0; i < n; i++)
        {
            Console.Write("Digite um número: ");
            x = int.Parse(Console.ReadLine()!);

            if (LOWER_BOUND <= x && x <= UPPER_BOUND)
            {
                insideCount++;
            }
            else
            {
                outsideCount++;
            }
        }

        Console.WriteLine($"{insideCount} DENTRO");
        Console.WriteLine($"{outsideCount} FORA");
    }
}

File: python/src/estruturas/repetitivas/dentro_fora.py
""" Problema "dentro_fora"
Leia um valor inteiro N.
Este valor será a quantidade de valores inteiros X que serão lidos em seguida.
Mostre quantos destes valores X estão dentro do intervalo [10,20] e quantos estão fora do intervalo.
"""

LOWER_BOUND = 10
UPPER_BOUND = 20

MIN_NUMBERS = 1
MAX_NUMBERS = 10


def dentro_fora() -> None:
    n = int(input(f"Quantos números você vai digitar (mínimo de {MIN_NUMBERS} e máximo de {MAX_NUMBERS})? "))

    if not MIN_NUMBERS < n < MAX_NUMBERS:
        n = MIN_NUMBERS

    out_of_range_numbers = 0
    in_range_numbers = 0

    for i in range(1, n + 1):
        x = int(input(f"Digite {i}º número de {n}: "))

        if LOWER_BOUND <= x <= UPPER_BOUND:
            in_range_numbers += 1
        else:
            out_of_range_numbers += 1

    print(f"Faixa: {LOWER_BOUND} <= x <= {UPPER_BOUND}")
    print(f"{in_range_numbers:2} DENTRO")
    print(f"{out_of_range_numbers:2} FORA")


if __name__ == "__main__":
    dentro_fora()
